src/components/UserContext/UserContext.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+
+type Context = {
+  todos: Todo[],
+  setTodos: (todosFromServer: Todo[]) => void,
+  selectedUser: number,
+  setSelectedUser: (selectedUser: number) => void,
+  selectedTodoId: number,
+  setSelectedTodoId: (selectedTodoId: number) => void,
+  userInfo: User | null,
+  setUserInfo: (userData: User | null) => void,
+  sortType: string,
+  setSortType: (type: string) => void,
+  query: string,
+  setQuery: (newQuery: string) => void,
+  appliedQuery: string,
+  setAppliedQuery: (finalQuery: string) => void,
+};
+
+export const UserContext = React.createContext<Context>({
+  todos: [],
+  setTodos: () => {},
```

Comment:
it's not great to move todos and todos handler to User Context
will be better to create separate TodoContext for this purpose

Reply:
will try to fix it on weekend